indexing-service/src/main/java/org/apache/druid/indexing/common/config/TaskConfig.java

```diff
@@ -292,6 +320,16 @@ public boolean isEncapsulatedTask()
     return encapsulatedTask;
   }
 
+  public synchronized void addTask(final String taskId, final File taskDir)
+  {
+    tracker.addTask(taskId, taskDir);
+  }
+
+  public synchronized void removeTask(final String taskId)
```

Comment:
I think `addTask` is not really needed anymore since `getBaseTaskDir` should automatically do the `addTask` part. Probably the `removeTask` could be made as `stopTracking(taskId)` if we remove `addTask`

Reply:
We do need addTask in the case of restore, which is why this is being left as it is